build.gradle

```diff
@@ -140,3 +140,50 @@ spotless {
         targetExclude 'src/test/java/org/mockitousage/internal/junit/UnusedStubbingsFinderTest.java'
     }
 }
+
+
+/*
+ * SourceSet for Java 17 tests
+ */
+sourceSets {
+    test17 {
```

Comment:
Instead of a new sourceset on the main project, let's define a separate subproject for this. Instead of setting to 17 (which is non-LTS), let's use 21 for this. E.g. create `subprojects/java21`. We can use it for #3167 as well

Reply:
As posted in #3167 let's create a separate to introduce a Java 21 subproject so that we can coordinate it (and not do double-work in these two PRs)